src/redux/superadmins/thunks.js

```diff
@@ -1,18 +1,92 @@
-import { getSuperadminsPending, getSuperadminsError, getSuperadminsSuccess } from './actions';
-
-export const getSuperAdmins = () => {
-  return async (dispatch) => {
-    try {
-      dispatch(getSuperadminsPending());
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/`);
-      const responseJson = await response.json();
-      const data = responseJson.data;
-      if (responseJson.error) {
-        throw new Error(responseJson.message);
+import * as actions from './actions';
+
+export const getSuperAdmins = async (dispatch) => {
+  dispatch(actions.getSuperadminsPending());
+  try {
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/`);
+    const data = await res.json();
+    dispatch(actions.resetInitialState());
+
+    if (res.status === 200) {
+      dispatch(actions.getSuperadminsSuccess(data.data));
+    }
+
+    if (res.status !== 200) {
+      throw new Error(data.message);
+    }
+  } catch (error) {
+    dispatch(actions.getSuperadminsError(error.message));
+  }
+};
+
+export const postSuperAdmins = async (dispatch, newSuperadmin) => {
+  try {
+    dispatch(actions.postSuperadminsPending());
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/`, {
+      method: 'POST',
+      body: JSON.stringify(newSuperadmin),
+      headers: {
+        'Content-Type': 'application/json'
       }
-      dispatch(getSuperadminsSuccess(data));
-    } catch (error) {
-      dispatch(getSuperadminsError(error.message));
+    });
+    const data = await res.json();
+    dispatch(actions.resetInitialState());
+
+    if (res.status === 201) {
+      dispatch(actions.postSuperadminsSuccess(data.data, data.message));
```

Comment:
You can destructure data as { data, message }, instead of data.data or data.message